### PR TITLE
[Windows, Keyboard] Lift key event redispatching to KeyboardManagerWin32

### PR DESCRIPTION
**Pull request by dkwingsmt**
This PR moves the mechanism that recognizes redispatched key events from `KeyboardKeyHandler`to `KeyboardManagerWin32`. Also, `TextInputPlugin` is now invoked during the callback of the original event, instead of the redispatched event.

image

Reasoning: 
- Event redispatching is only available on Win32, not on UWP, therefore everything after `FlutterWindowView` is now (almost) platform agnostic, and the redispatching module is Win32-aware, allowing it to perform Win32-specific operations.
- Redispatched events will no longer reach `KeyboardKeyHandler` and `TextInputPlugin`.
- `TextInputPlugin` now receives the original events, instead of redispatched events. This has been the suspect for various bugs.

Some tests have been changed:
- All `OnText` calls now occur on the original event instead of the redispatched event, breaking some tests. This is expected and should not affect user experience at all.
- High surrogate events are now always passed to the system. This is temporary, and will be fixed in the (near) future when reworking the redispatching logic. For now this only affects an extremely rare range of keyboards (such as Gothic) while using add-to-app.

Other than the broken tests, this PR should not need new tests since it's a refactor.

This is one of the many steps to [refactor the windows keyboard system](https://github.com/flutter/flutter/issues/88021).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/testing/test_keyboard.cc [hunk 1]
```diff
@@ -38,6 +38,19 @@ std::unique_ptr<std::vector<uint8_t>> _keyHandlingResponse(bool handled) {
   return flutter::JsonMessageCodec::GetInstance().EncodeMessage(document);
 }
 
+static std::string ordinal(int num) {
```
Comment (gspencergoog) [inline shell/platform/windows/testing/test_keyboard.cc:41]: Nice touch!

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: shell/platform/windows/flutter_window_winuwp.cc [hunk 2]
```diff
@@ -67,6 +67,10 @@ winrt::Windows::UI::Core::CoreCursor GetCursorByName(
   }
 }
 
+// UWP does not care if key events are handled, since it can not redispatch them
```
[the remark this answers is not in the document]
Comment (dkwingsmt) [reply shell/platform/windows/flutter_window_winuwp.cc:70]: UWP can't redispatch events at all, so it's equivalent to handling all events (outside native components can't receive any events.) Or we can always pass events to native components.

[the remark this answers is not in the document]
Comment (gspencergoog) [reply shell/platform/windows/flutter_window_winuwp.cc:70]: Next time you're in the code, could you please add a comment about this somewhere appropriate: just that the default is to handle all events, and why?

**Pull request merged**